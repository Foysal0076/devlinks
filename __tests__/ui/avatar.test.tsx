import React from 'react'
import { render, screen } from '@testing-library/react'
import Avatar from '@/components/ui/avatar'

// Mock the next/image component
jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => <img {...props} />,
}))

describe('Avatar component', () => {
  test('renders with initials when no URL is provided', () => {
    render(<Avatar name='John Doe' />)
    expect(screen.getByText('JD')).toBeInTheDocument()
  })

  test('applies custom className to the container', () => {
    const { container } = render(
      <Avatar name='John Doe' className='custom-class' />
    )
    expect(container.firstChild).toHaveClass('custom-class')
  })

  test('applies custom textClassName to the initials', () => {
    render(<Avatar name='John Doe' textClassName='custom-text-class' />)
    expect(screen.getByText('JD')).toHaveClass('custom-text-class')
  })

  test('uses createInitial function correctly', () => {
    render(<Avatar name='Alice Bob Charlie' />)
    expect(screen.getByText('AB')).toBeInTheDocument()
  })

  test('renders with default classes', () => {
    const { container } = render(<Avatar name='John Doe' />)
    expect(container.firstChild).toHaveClass(
      'relative',
      'flex',
      'h-8',
      'w-8',
      'items-center',
      'justify-center',
      'rounded-full',
      'border-neutral-100',
      'bg-surface-300',
      'text-base',
      'font-bold'
    )
  })
  test('matches snapshot', () => {
    const { asFragment } = render(<Avatar name='John Doe' />)
    expect(asFragment()).toMatchSnapshot()
  })
})
