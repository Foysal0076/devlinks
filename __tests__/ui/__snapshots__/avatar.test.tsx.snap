// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative flex h-8 w-8 items-center justify-center rounded-full border-neutral-100 bg-surface-300 text-base font-bold"
  >
    <span
      class="text-sm font-bold md:text-base"
    >
      JD
    </span>
  </div>
</DocumentFragment>
`;
